### ISSUE TITLE
Issues with FW8

### ISSUE THREAD

**Issue by heruan** [bdunn44/FontAwesome-Vaadin#5]
Does it work on Vaadin Framework 8.0.0.beta1? I tried adding the dependency and then:

```java
import com.kbdunn.vaadin.addons.fontawesome.FontAwesome;

Label test = new Label("Test");
test.setIcon(FontAwesome.USER);
```

but I get a square icon:

![screen shot 2017-01-17 at 12 43 24](https://cloud.githubusercontent.com/assets/851745/22019187/bbd99c9c-dcb2-11e6-8af5-6579d71310dd.png)

**Comment by bdunn44** [comment]
I have not tested with Vaadin 8 - if you find the source of the problem please post it here

**Comment by nikkijuk** [comment]
in Vaadin 8 Beta2 FontAwesome is marked as deprecated in favor to vaadins own icons, which might mean that this addon will come pretty popular if users of FW8 are wanting still to use FontAwesome with recent set of icons.

**Comment by nikkijuk** [comment]
I did test icons with FW8 beta 2 grid (example with all checked value provider: addColumn(row -> FontAwesome.CHECK.getHtml(), new HtmlRenderer()) ..) and it worked just as expected. I feel that it might be good to run some more tests and then declare in vaadin add on repository that this add on is vaadin 8 compatible. I guess, but haven't tested, that label would also work, but label content needs to be defined to be html (example from vaadin 7.2 docs): label.setContentMode(ContentMode.HTML);

**Comment by heruan** [comment]
I got it working in FW8 too, it was just the `import` in `styles.scss` wrong (the README need an update)!

```scss
@import "../../addons/font-awesome-label/font-awesome";

.my-theme {
    @include font-awesome;
}

```

**Comment by heruan** [comment]
Too fast, ouch! With FW 8.0.0.beta2 the plugin won't work, since `ContentMode` was moved from `com.vaadin.shared.ui.label` to `com.vaadin.shared.ui`.

**Comment by nikkijuk** [comment]
True. Moving of ContentMode made my imports to break, but otherwise move from Beta1 -> Beta2 was smooth. Does this mean that there needs to be separate version for FW8?

**Comment by bdunn44** [comment]
Yes it would need a separate release for that very small fix. I'll work on that and further testing when I get a few minutes. PRs are always welcome!

**Comment by heruan** [comment]
This PR simply update the package, not the dependencies as I do not know Ivy: https://github.com/bdunn44/FontAwesome-Vaadin/pull/6

On my fork, I have a `fw8` branch were I have restructured the plugin for Maven following Vaadin archetype structure (addon + demo): https://github.com/heruan/FontAwesome-Vaadin/tree/fw8

@bdunn44 Do you think you'll stick with Ivy for future releases? FontAwesome 5 is coming soon 😃

**Comment by bdunn44** [comment]
Fixed with d63ee3c79aeb620a0a5c476a6f44f7855d2eb345 and release 1.4.0!
